server/api/notification.go

```diff
@@ -184,49 +194,153 @@ func (h *notificationHandler) processNotification(n *remote.Notification) error
+			Short: true,
+		})
 	}
 	for _, k := range updated {
-		message += fmt.Sprintf("- %s: ~~%s~~ \u2192 %s\n",
-			k, priorFields[k].Strings(), newFields[k].Strings())
+		sa.Fields = append(sa.Fields, &model.SlackAttachmentField{
+			Title: k,
+			Value: fmt.Sprintf("~~%s~~ \u2192 %s", priorFields[k].Strings(), newFields[k].Strings()),
+			Short: true,
+		})
 	}
 	for _, k := range deleted {
-		message += fmt.Sprintf("- %s: ~~%s~~\n", k, newFields[k].Strings())
+		sa.Fields = append(sa.Fields, &model.SlackAttachmentField{
+			Title: k,
+			Value: fmt.Sprintf("~~%s~~", priorFields[k].Strings()),
+			Short: true,
+		})
 	}
 
-	h.Logger.LogDebug("Processed event notification",
-		"SubsriptionID", n.SubscriptionID,
-		"Message", message)
-	return true, message
+	h.addPostActionSelect(sa, n.Event)
+	return true, sa
 }
 
-func (h *notificationHandler) formatNewEventNotification(n *remote.Notification) string {
-	message := ""
-	if n.ChangeType == "created" {
-		message = fmt.Sprintf("New event: [%s](%s)\n", n.Event.Subject, n.Event.Weblink)
-	} else {
-		message = fmt.Sprintf("Previously unseen event: [%s](%s)\n", n.Event.Subject, n.Event.Weblink)
+func (h *notificationHandler) actionURL(action string) string {
+	return fmt.Sprintf("%s/%s/%s", h.Config.PluginURLPath, config.PathPostAction, action)
+}
+
+func (h *notificationHandler) addPostActions(sa *model.SlackAttachment, event *remote.Event) {
+	if !event.ResponseRequested {
+		return
+	}
+	context := map[string]interface{}{
+		config.EventIDKey: event.ID,
 	}
-	for k, v := range eventToFields(n.Event) {
-		message += fmt.Sprintf("- %s: %s\n", k, v.Strings())
+	sa.Actions = []*model.PostAction{
+		{
+			Name: "Accept",
+			Type: model.POST_ACTION_TYPE_BUTTON,
+			Integration: &model.PostActionIntegration{
+				URL:     h.actionURL(config.PathAccept),
+				Context: context,
+			},
+		},
+		{
+			Name: "Tentatively Accept",
+			Type: model.POST_ACTION_TYPE_BUTTON,
+			Integration: &model.PostActionIntegration{
+				URL:     h.actionURL(config.PathTentative),
+				Context: context,
+			},
+		},
+		{
+			Name: "Decline",
+			Type: model.POST_ACTION_TYPE_BUTTON,
+			Integration: &model.PostActionIntegration{
+				URL:     h.actionURL(config.PathDecline),
+				Context: context,
+			},
+		},
 	}
-	return message
+}
+
+func (h *notificationHandler) addPostActionSelect(sa *model.SlackAttachment, event *remote.Event) {
+	if !event.ResponseRequested {
+		return
+	}
+
+	context := map[string]interface{}{
+		config.EventIDKey: event.ID,
+	}
+
+	pa := &model.PostAction{
+		Name: "Response",
+		Type: model.POST_ACTION_TYPE_SELECT,
+		Integration: &model.PostActionIntegration{
+			URL:     h.actionURL(config.PathRespond),
+			Context: context,
+		},
+	}
+
+	for _, o := range []string{OptionNotResponded, OptionYes, OptionNo, OptionMaybe} {
+		pa.Options = append(pa.Options, &model.PostActionOptions{Text: o, Value: o})
+	}
+	switch event.ResponseStatus.Response {
+	case "notResponded":
```

Comment:
Not blocking but can these be constantized?

Reply:
Will do separately, more changes are likely to come to that interface.